include/git2/remote.h

```diff
@@ -386,6 +386,18 @@ typedef struct {
 typedef int (*git_push_negotiation)(const git_push_update **updates, size_t len, void *payload);
 
 /**
+ * Called for each updated reference on push. If `status` is
+ * not `NULL`, the update was rejected by the remote server
+ * and `status` contains the reason given.
```

Comment:
The style for function documentation is usually as follows:

```
/**
 * Short summary of that thing
 *
 * A longer description which may span multiple lines
 * and describes what that symbol actually is/does.
 *
 * @param frobnicate frobnicate all the stuff
 * @return 0 on foo, otherwise an error
 */
```

So please separate the summary and descriptions here and add an `@return` description, such that developers know what they may return from that callback.